### PR TITLE
[TorchFX] Conformance test init

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

Conformance test for resnet18

### Reason for changes

To extend testing scope for the TorchFX backend

### Related tickets

https://github.com/openvinotoolkit/nncf/issues/2766

### Tests

post_training_quantization/442 is successfull

### CODE REVIEW

**Comment by daniil-lyakhov** [comment]
Build post_training_quantization/430 is successfull

File: tests/post_training/test_quantize_conformance.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def test_ptq_quantization(
     err_msg = None
     test_model_param = None
     start_time = time.perf_counter()
-    try:
+    if True:
```
Comment (AlexanderDokuchaev) [inline tests/post_training/test_quantize_conformance.py:250]: Looks like used for debug

File: tests/post_training/data/ptq_reference_data.yaml [hunk 2]
```diff
@@ -22,6 +22,16 @@ hf/hf-internal-testing/tiny-random-gpt2_backend_OV:
   metric_value: null
 hf/hf-internal-testing/tiny-random-gpt2_backend_TORCH:
   metric_value: null
+torchvision/resnet18_backend_FP32:
```
Comment (AlexanderDokuchaev) [inline tests/post_training/data/ptq_reference_data.yaml:25]: What differ between torchvision and timm models?
If it's same but timm can not be exported to FX, timm/resnet18 can be removed

File: tests/post_training/pipelines/base.py [hunk 3]
```diff
@@ -367,6 +368,11 @@ def save_compressed_model(self) -> None:
             )
             self.path_compressed_ir = self.output_model_dir / "model.xml"
             ov.serialize(ov_model, self.path_compressed_ir)
+        elif self.backend == BackendType.FX_TORCH:
+            exported_model = torch.export.export(self.compressed_model, (self.dummy_tensor,))
```
Comment (AlexanderDokuchaev) [inline tests/post_training/pipelines/base.py:372]: Compressed FX model can not be converted to OV model directly?

File: tests/post_training/pipelines/image_classification_torchvision.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        return transform_fn
+
+    def prepare_calibration_dataset(self):
+        dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
+        loader = torch.utils.data.DataLoader(dataset, batch_size=self.batch_size, num_workers=2, shuffle=False)
+
+        self.calibration_dataset = nncf.Dataset(loader, self.get_transform_calibration_fn())
+
+    def _validate(self):
+        val_dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
```
Comment (AlexanderDokuchaev) [inline tests/post_training/pipelines/image_classification_torchvision.py:126]: Looks like it can be reused from image_classification_timm.py

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

File: tests/post_training/pipelines/base.py (hunk 3 above)
Comment (daniil-lyakhov) [reply tests/post_training/pipelines/base.py:372]: Yes, the error is
```
 Failed: Couldn't get TorchScript module by tracing. With exception:
E           
E                   Calling train() or eval() is not supported for exported models.
E                   Please call `torch.ao.quantization.move_exported_model_to_train(model)` (or eval) instead.
E           
E                   If you cannot replace the calls to `model.train()` and `model.eval()`, you may override
E                   the behavior for these methods by calling `torch.ao.quantization.allow_exported_model_train_eval(model)`,
E                   which does the above automatically for you. Note that this has limited effect on switching
E                   behavior between train and eval modes, and should be used only for special ops such as dropout
E                   and batchnorm.

```

File: tests/post_training/pipelines/image_classification_torchvision.py (hunk 4 above)
Comment (daniil-lyakhov) [reply tests/post_training/pipelines/image_classification_torchvision.py]: Done

File: tests/post_training/data/ptq_reference_data.yaml (hunk 2 above)
Comment (daniil-lyakhov) [reply tests/post_training/data/ptq_reference_data.yaml:25]: I've removed timm/resnet18 and replaced with torchvision/resnet18

File: tests/post_training/test_quantize_conformance.py (hunk 1 above)
Comment (daniil-lyakhov) [reply tests/post_training/test_quantize_conformance.py:250]: My bad, done!

**Comment by daniil-lyakhov** [comment]
Build post_training_quantization/442 is green

**Review by AlexanderDokuchaev — APPROVED** [review]

**Pull request merged**